src/controllers/importer/ImporterController.php

```diff
@@ -0,0 +1,134 @@
+<?hh // strict
+
+abstract class ImporterController {
+  abstract public function importData(string $input_filename, string $type): ?array<string, mixed>;
+
+  public async function processTeams(array<string, mixed> $teams): Awaitable<bool> {
```

Comment:
Maybe the argument type should be `array<string, array<string, mixed>>`? That also might fix your hack errors. Same argument for the other functions.